libraries/standard/mqtt/src/iot_mqtt_network.c

```diff
@@ -274,11 +315,155 @@ static IotMqttError_t _getIncomingPacket( void * pNetworkConnection,
 
 /*-----------------------------------------------------------*/
 
+static IotMqttError_t _deserializeAck( _mqttConnection_t * pMqttConnection,
+                                       _mqttPacket_t * pIncomingPacket,
+                                       IotMqttDeserialize_t _deserializer,
+                                       IotMqttOperationType_t opType,
+                                       uint16_t * pPacketIdentifier )
+{
+    IotMqttError_t status = IOT_MQTT_STATUS_PENDING;
+    _mqttOperation_t * pOperation = NULL;
+
+    status = _deserializer( pIncomingPacket );
+
+    pOperation = _IotMqtt_FindOperation( pMqttConnection,
+                                         opType,
+                                         pPacketIdentifier );
+
+    if( pOperation != NULL )
+    {
+        pOperation->u.operation.status = status;
+        _IotMqtt_Notify( pOperation );
+    }
+
+    return status;
+}
```

Comment:
This function was abstracted out since four very similar blocks of code existed in `_deserializeIncomingPacket`, and so that each case in the `switch` statement would look more consistent. However, most of the complexity came from the publish and pingresp cases. Even without this function, the complexity of `_deserializeIncomingPacket` is still below 8, so it isn't absolutely necessary.